### PR TITLE
Fixes following jordigh review

### LINKED ISSUE #87: Installation on OSX documentation

### LINKED ISSUE #98: constexpr void is not a C++11 feature
The following test is incorrect:

https://github.com/artivis/manif/blob/joss/CMakeLists.txt#L34

This is because this was only allowed by C++14:

https://isocpp.org/files/papers/N3652.html

Both clang and gcc fail this test with -std=c++11

### LINKED ISSUE #99: Build instructions somewhat unclear about which make steps to run
The library is header-only. This means that there is nothing to build at installation time.

The documentation suggests running `make` after `cmake`, but unless we are building the tests and examples, this is pointless.

Morever, the build instructions do not indicate a `make install` step. This is necessary so that manif will be found according to the instructions in the "Use Manif in your project" section of the Readme.

### PR DESCRIPTION
**Pull request by artivis**
- remove `constexpr void` function feature altogether as it is used in a single place and tedious to support on different compilers
- OS X install eigen
- slightly fix build instructions
- various readme fixes, typo etc 

Fix #87 
Fix #98 
Fix #99 
Replace #100

### CODE REVIEW

**Comment by artivis** [comment]
@jordigh I'll answer your review in this PR.

File: README.md [hunk 1]
```diff
@@ -80,14 +80,11 @@ $ apt-get install manif
 ```terminal
 $ git clone https://github.com/artivis/manif.git
 $ cd manif && mkdir build && cd build
-$ cmake ..
-$ make
-```
-###### To build also examples/tests
-```terminal
 $ cmake -DBUILD_TESTING=ON -DBUILD_EXAMPLES=ON ..
 $ make
```
Comment (joansola) [inline README.md]: It appears 'make' alone does nothing no? Or is it to make tests and examples?

Comment (artivis) [reply README.md]: Both tests and examples build are enabled thus `make` should build both.

**Comment by artivis** [comment]
CI fails for Clang, I'll have to look further on how to properly detect compile features or simply discard all together. This seems like a lot of trouble for a very small feature.

**Comment by joansola** [comment]
It fails in my mac. Let me know if you want the make log.

**Comment by artivis** [comment]
@joansola I know it is failing for Clang, I have the log on TravisCI. I'll fix it asap.

**Comment by artivis** [comment]
@jordigh I have cherry-picked your change in this PR (from #100) and reverted what I meant to revert. If you agree I'll close #100 in favor of this one.

**Comment by joansola** [comment]
I confirm correct compilation on Mac using clang now.

**Comment by artivis** [comment]
@joansola Thanks. I'll wait a bit giving @jordigh a chance to see this PR before merging.

**Comment by jordigh** [comment]
Sure, it's your project, you don't need my permission to modify its code. :-)

**Comment by artivis** [comment]
@jordigh Ahah no I just wanted you to see those comments before you receive a message that I closed your PR #100.  
Closing and merging then :+1: .

**Pull request merged**